odoo/tools/template_inheritance.py

```diff
@@ -205,23 +205,94 @@ def extract(spec):
                     raise ValueError(_("Invalid mode attribute:") + " '%s'" % mode)
             elif pos == 'attributes':
                 for child in spec.getiterator('attribute'):
-                    attribute = child.get('name')
-                    value = child.text or ''
-                    if child.get('add') or child.get('remove'):
-                        assert not child.text
-                        separator = child.get('separator', ',')
-                        if separator == ' ':
-                            separator = None    # squash spaces
-                        to_add = (
-                            s for s in (s.strip() for s in child.get('add', '').split(separator))
-                            if s
+                    attribute = child.attrib.pop('name')
+                    value = None
+
+                    if child.get('add') or child.get('remove') or 'separator' in child.attrib:
+                        if child.text:
+                            raise ValueError(
+                                _("Forbidden text %s") %
+                                repr(child.text)
+                            )
+                        value = node.get(attribute, '')
+
+                        if attribute in ('invisible', 'column_invisible', 'required', 'readonly', 't-if', 't-elif') or attribute.startswith('decorator-'):
+                            # python expression
+                            value = f' ({value}) ' if value else value
```

Comment:
This is making everything more complicated.  The add or remove operation is done on `value`, not `f" ({value}) "`.